src/platform/win32bufferedfile.cpp

```diff
@@ -0,0 +1,290 @@
+        modestr.Concat('b');
+    }
+
+    // #TODO Treat filename as UTF8 and call wide char version of fopen?
+
+    m_file = fopen(filename, modestr.Str());
+
+    if (m_file == nullptr) {
+        return false;
+    }
+
+    ++s_totalOpen;
+
+    if ((m_openMode & FileMode::APPEND) != 0 && Seek(0, SeekMode::END) < 0) {
+        Close();
+        return false;
+    }
+
+    if (buffer_size != 0) {
+        if (0 != setvbuf(m_file, nullptr, _IOFBF, buffer_size)) {
+            captainslog_dbgassert(0, "File buffer could not be set");
+        }
+    } else {
+        // Uses default buffer size - likely 512.
+    }
+
+    return true;
+}
+
+int Win32BufferedFile::Read(void *dst, int bytes)
+{
+    if (!m_access) {
+        return -1;
+    }
+
+    if (dst != nullptr) {
+        return fread(dst, 1, bytes, m_file);
+    } else {
+        Seek(bytes, SeekMode::CURRENT);
+    }
+
+    return bytes;
+}
+
+int Win32BufferedFile::Write(void const *src, int bytes)
+{
+    if (!m_access || src == nullptr) {
+        return -1;
+    }
+
+    return fwrite(src, 1, bytes, m_file);
+}
+
+int Win32BufferedFile::Seek(int offset, File::SeekMode mode)
+{
+    int fmode;
+    switch (mode) {
+        case SeekMode::START:
+            fmode = SEEK_SET;
+            break;
+        case SeekMode::CURRENT:
+            fmode = SEEK_CUR;
+            break;
+        case SeekMode::END:
+            fmode = SEEK_END;
+            break;
+        default:
+            return -1;
+    }
+
+    return fseek(m_file, (long)offset, fmode);
+}
+
+void Win32BufferedFile::Next_Line(char *dst, int bytes)
+{
+    captainslog_trace("Seeking getting next line from Win32BufferedFile %s.", m_filename.Str());
+
+    int i;
+
+    for (i = 0; i < bytes - 1; ++i) {
+        char tmp;
+        size_t ret = fread(&tmp, sizeof(tmp), 1, m_file);
+
+        if (ret == 0 || tmp == '\n') {
+            break;
+        }
+
+        if (dst != nullptr) {
+            dst[i] = tmp;
+        }
+    }
+
+    if (dst != nullptr) {
+        dst[i] = '\0';
+    }
+}
+
+bool Win32BufferedFile::Scan_Int(int &integer)
+{
+    captainslog_trace("Scanning Int from Win32BufferedFile %s.", m_filename.Str());
+    char tmp;
+    Utf8String number;
+
+    integer = 0;
+
+    // Loop to find the first numeric character.
+    do {
+        if (fread(&tmp, sizeof(tmp), 1, m_file) == 0) {
+            return false;
+        }
+    } while (!isdigit(tmp) && tmp != '-');
+
+    // Build up our string of numeric characters
+    while (true) {
+        number.Concat(tmp);
+
+        if (fread(&tmp, sizeof(tmp), 1, m_file) == 0) {
+            break;
+        }
+
+        if (!isdigit(tmp)) {
+            // If we read a byte, seek back that byte for the next read
+            // as we are done with the current number.
+            Seek(-1, SeekMode::CURRENT);
+
+            break;
+        }
+    }
+
+    integer = atoi(number.Str());
+
+    return true;
+}
+
+bool Win32BufferedFile::Scan_Real(float &real)
```

Comment:
I copied all functions from `Win32LocalFile` as a base, so the algorithms are an exact match.